Documentation (#131)
PM-Dungeon/core: README.md

> # PM-Dungeon-Framework

Reviewer: irgendwas mit "Core" noch im Namen?

Author: so irgendwie gemacht, schaus dir mal an.

Reviewer: Hmmm, das passt für mich :)

> 

Reviewer: ## The unique Rougelike Game

Ein paar Sätze über das Framework ...

Author: Hab überleg die Repo Beschreibung da rein zu kopieren aber irgendwie ist das so doppelt gemoppelt.

Reviewer: weiss nicht. andere machen hier tatsächlich einen kurzen absatz, was man tolles mit dem projekt anfangen kann. die repo-beschreibung ist ja nur ein satz und quasi die essenz aus dem text hier ...

Author: ich überleg mir mal was schlaues....

> To create your own game with the PM Dungeon Framework, check out the Quickstart Guide: 

Reviewer: ```suggestion
## Installing & Quickstart

To create your own game with the PM-Dungeon-Framework, check out the Quickstart Guide: 
```

> - German **KOMMT NOCH**

Reviewer: ```suggestion
- German **TODO**
```

Author: ist nur nen vermerk für mich/euch. Wird so nicht gemerged. Ich trag da dann die entsprechenden links ein.

> - English **KOMMT NOCH**

Reviewer: ```suggestion
- English **TODO**
```

## What do you need

> - JDK v.17.0 or higher

Reviewer: bei Oracle heisst das "Java 17 LTS", "Java SE Development Kit 17.0.x" ... (https://www.oracle.com/java/technologies/downloads/#jdk17-mac) ... Sollten wir das auch irgendwie so benennen?

Und ist das wirklich alles? Braucht man nicht noch mehr?

Author: och oracle.... ja wir sollten uns vermutlich danach richten.

> Und ist das wirklich alles? Braucht man nicht noch mehr?

Ich wüsste im Grunde nicht was. gradlew liefern wir mit und coden kannste im texteditor :P

Reviewer: naja, wir brauchen noch desktop und libgdx. okay, das ist irgendwie eine andere liga (das erste kann man quasi selsbt erstellen, das letztere zieht sich unser gradle). hmmm.

aber bei der java-versionsbezeichnung sollten wir imho schon die "offizielle" bezeichnung nehmen :)

> ## How to contribute

Reviewer: ```suggestion
## Contributing
```

> 

Reviewer: ```suggestion

Pull requests, bug reports, and feature requests are welcome. Please make sure to read the [contributor guidelines](CONTRIBUTING.md) before opening a new issue.
```

> If you would like to contribute to the project, please follow the code convention. 

Reviewer: ab hier "If you would ... " bis "... to your support." würde ich alles in die `CONTRIBUTING.md` packen.

If you use the IntelliJ IDEA you can use our [Configuration Guide](https://github.com/PM-Dungeon/core/wiki/Codeformatter-f%C3%BCr-IntelliJ-IDEA-konfigurieren) to set up your project accordingly. Please use the appropriate templates when creating an issue or pull request. 

We look forward to your support.

> ## Language definition  

Reviewer: ```suggestion
## Language  
```

> This project is implemented by a German-speaking university and is therefore designed for German-speaking students. Nevertheless, the project can also be used by international institutes or private individuals. Please note, however, that the English documentation may not be available or may be outdated. If you have any questions, problems or suggestions, please feel free to contact us in English. 

Reviewer: ```suggestion
This project is intended as teaching material for German-language university courses and is therefore aimed at German-speaking students. Please keep in mind that the English documentation may therefore not be available or may be slightly out of date. If you have any questions, problems or suggestions, please feel free to contact us in English or German. 
```

> 

Reviewer: ```suggestion
## License

Copyright (c) 2021-2022 André Matutat, Carsten Gips. Released under the [MIT](LICENSE.md) license.
```

This is the `core` part of the project. It contains the actual framework without any launcher implementation. You can use it to develop your own game, but it is probably easier to use the [desktop](https://github.com/PM-Dungeon/desktop) project as a starting point for your own game. 

## Requirements

- Java SE Development Kit 17.0.x or higher

## Installing & quick start

To create your own game using the PM-Dungeon framework, check out the quick start guide: 

- [German](./documentation/quickstart_de.md)
- [English](./documentation/quickstart_en.md)

## Contributing

Pull requests, bug reports, and feature requests are welcome. Please make sure to read the [contributor guidelines](CONTRIBUTING.md) before opening a new issue.

## Language  

This project is intended as teaching material for German-language university courses and is therefore aimed at German-speaking students. Please keep in mind that the English documentation may therefore not be available or may be slightly out of date. If you have any questions, problems or suggestions, please feel free to contact us in English or German. 

## License

> See [License](./LICENSE.md)

Reviewer: Ich fand das alte eigentlich besser, auch wenn es sich doppelt. Andere Projekte machen das auch so. Für Leser wird dann sofort klar, welche Lizenz und wer, ohne dass man noch in die Datei rein muss. Dafür würde ich die Doppelung in Kauf nehmen.

# PM-Dungeon-Framework

## About

> The **PM-Dungeon** is a lightweight, [libgdx](https://libgdx.com/) based, Java framework for the development of a 2D rogue like. It is developed for teaching purposes and therefore especially suitable for inexperienced Java developers. The project is free to use for your own lecture.

Reviewer: ```suggestion
The **PM-Dungeon** is a lightweight Java framework for the development of a 2D-Rougelike game. It is designed for educational purposes and is especially aimed at inexperienced Java developers. The project currently uses the [libGDX](https://libgdx.com/) as a backend.
```

> This is the `core` part of the project. In contains the framework itself without any launcher implementation. You can use it with your own `libGDX`-project, or you can use the provided [desktop-setup](https://github.com/PM-Dungeon/desktop). 

Reviewer: ```suggestion
This is the `core` part of the project. It contains the actual framework without any launcher implementation. You can use it to develop your own game, but it is probably easier to use the [desktop](https://github.com/PM-Dungeon/desktop) project as a starting point for your own game. 
```

> ## What do you need

Reviewer: ```suggestion
## Requirements
```

Reviewer: In den anderen Überschriften haben wir das neutral formuliert, das sollten wir hier dann auch tun.

- Java SE Development Kit 17.0.x or higher

## Installing & quick start

> To create your own game with the PM-Dungeon-Framework, check out the quick start guide: 

Reviewer: ```suggestion
To create your own game using the PM-Dungeon framework, check out the quick start guide: 
```

> - [German](./documentation/qucikstart_de.md)

Reviewer 2: Hier hat sich ein typo eingeschlichen =) qucikstart klingt schön xD
```suggestion
- [German](./documentation/quickstart_de.md)
```

> - [English](./documentation/qucikstart_de.md)

Reviewer 2: ```suggestion
- [English](./documentation/quickstart_en.md)
```

> If you just want to get started, add `implementation "io.github.pm-dungeon:core:1.0.+"` to your dependencies.

Reviewer: Der Satz ist unvollständig. Wo genau soll man das hinzufügen? Was kann man dann starten? ...

Ich würde diesen Satz hier entfernen und in die Quickstart-Anleitungen aufnehmen.

Author: done.
Ist da auch schon drinnen.

## Contributing

Pull requests, bug reports, and feature requests are welcome. Please make sure to read the [contributor guidelines](CONTRIBUTING.md) before opening a new issue.

## Language  

This project is intended as teaching material for German-language university courses and is therefore aimed at German-speaking students. Please keep in mind that the English documentation may therefore not be available or may be slightly out of date. If you have any questions, problems or suggestions, please feel free to contact us in English or German. 

## License

> Copyright (c) 2021-2022 André Matutat, Carsten Gips. Released under the [MIT](LICENSE.md) license.

Reviewer: @AMatutat hatten wir nicht grad erst einen PR, um @Tobias321 hier mit aufzunehmen? wo ist das hin?

Author: Der Branch ist älter als das Lizenz update. Ich frage mich nur, warum das hier als Änderung aufgeführt wird. 
- ggf. rebasen

Reviewer: ic. wir müssen definitiv schneller werden bzw. kleinere pr's machen.

und alte branches nochmal rebasen, bevor wir die arbeit darin wieder aufnehmen (wenn das dann doch mal ne weile rumlag).

Author: Generell ja, aber hier ist es anders.
Wir haben Tobi in die `LICENSE.md` aufgenommen, nicht in die `README`.
Hier sollte ein Link eingefügt werden, der auf die `LICENSE.MD` verweist, dann müssen wir das nicht doppelt pflegen.

Author: done
